### ISSUE TITLE
[Bug]: Resources cannot start if the name is too long

### ISSUE THREAD

**Issue by lucferbux** [opendatahub-io/odh-dashboard#1552]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Deploy type

OpenDataHub core version (eg. `v1.6.0`)

### Version

1.7.0

### Current Behavior

Similar issue as #1308 

If the combined length of the project name and other resource such as **workbenches** and **model servers** exceeds 63 characters, OCP will be unable to create the corresponding route for the inference endpoint (which takes the form of `<resrouceName>-<projectName>.xyz.foo.com`)

We are waiting on https://github.com/opendatahub-io/sig-ml-developer-experience/issues/8 to get a solution for this.

Right now is happening for:

- Pipelines
- Workbenches
- Model serving

### Expected Behavior

We can create the resources successfully.

### Steps To Reproduce

1. Create a DS Project
2. Deploy a resource  (make sure the name + project name length > 63 characters)

**Comment by andrewballantyne** [comment]
> Similar issue as https://github.com/opendatahub-io/odh-dashboard/issues/1308

Any reason we didn't just rework the other ticket @lucferbux ? What value are we getting here with the separation?

**Comment by lucferbux** [comment]
> > Similar issue as #1308
> 
> Any reason we didn't just rework the other ticket @lucferbux ? What value are we getting here with the separation?

Agh I'm sorry, I did it at EOD yesterday and I misread something, I thought it was actively being in progress right now, so I split it up in two.
Im gonna close this one and rework the other one, thanks @andrewballantyne
